tests/sentry/utils/performance_issues/test_performance_detection.py

```diff
@@ -922,6 +932,54 @@ def test_detects_slow_span_in_solved_n_plus_one_query(self):
             ]
         )
 
+    def test_detects_consecutive_db_spans(self):
+        SECOND = 1000
+        span_duration = 5 * SECOND
+        spans = [
+            create_span("db", span_duration, "SELECT `customer`.`id` FROM `customers`"),
+            create_span("db", span_duration, "SELECT `order`.`id` FROM `books_author`"),
+            create_span("db", span_duration, "SELECT `product`.`id` FROM `products`"),
+        ]
+        spans = list(
+            map(
+                lambda span: modify_span_start(span, span_duration * spans.index(span)), spans
+            )  # ensures spans don't overlap
+        )
+        consecutive_db_event = create_event(
+            spans,
+            "a" * 16,
+        )
+
+        sdk_span_mock = Mock()
+
+        _detect_performance_problems(consecutive_db_event, sdk_span_mock)
+        sdk_span_mock.containing_transaction.set_tag.assert_has_calls(
+            [
+                call("_pi_all_issue_count", 4),
+                call("_pi_sdk_name", "sentry.python"),
+                call("_pi_transaction", "aaaaaaaaaaaaaaaa"),
+                call("_pi_consecutive_db_fp", "1-GroupType.PERORMANCE_CONSECUTIVE_DB_OP"),
+                call(
+                    "_pi_consecutive_db",
+                    {
+                        "trace_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+                        "parent_span_id": "aaaaaaaaaaaaaaaa",
+                        "span_id": "bbbbbbbbbbbbbbbb",
+                        "start_timestamp": 0.0,
+                        "timestamp": 5.0,
+                        "same_process_as_parent": True,
+                        "op": "db",
+                        "description": "SELECT `customer`.`id` FROM `customers`",
+                        "fingerprint": None,
+                        "tags": None,
+                        "data": None,
+                        "hash": "",
+                    },
+                ),
+                call("_pi_slow_span", "bbbbbbbbbbbbbbbb"),
+            ]
```

Comment:
I'm not quite sure if this is the best way to test this. Because _detect_performance_problems did not return anything even tho _stored_problems did contain the issue, this means we can't test it's return value correctly. Didn't dive into this so much, but it seems that its because its not a `allowed_perf_issue_detectors`
https://github.com/getsentry/sentry/blob/9f23c75e5a101407df72917cc98c755c26522aba/src/sentry/utils/performance_issues/performance_detection.py#L268

Regardless, the goal here is to run this in detect only mode

Reply:
Ah, yes you're correct. Until actual issues are returned I think this is the best way we have to test (the other tests do similarly). It's fine for now.